MidiTools/src/main/kotlin/com/example/android/miditools/MidiCiInitiator.kt

```diff
@@ -186,6 +302,38 @@ class MidiCiInitiator {
         return outputMessage
     }
 
+    private fun waitForGrouplessMessage(midiReceiver: WaitingMidiReceiver, timeoutMs : Long)
+            : ByteArray {
+        val startTime = LocalDateTime.now()
+        var currentTime = startTime
+        val endTime = startTime.plus(timeoutMs, ChronoField.MILLI_OF_DAY.baseUnit)
+        var outputMessage = ByteArray(0)
+        var isGroupless = false
+        while (endTime > currentTime) {
+            val remainingDurationMs = ChronoUnit.MILLIS.between(currentTime, endTime)
+            midiReceiver.waitForMessages(midiReceiver.readCount + 1,
```

Comment:
Does this allow other unrelated messages, like NoteOffs to be handled?
We may need a chain of handlers. This could be a later CL.

Reply:
Added some comments. Currently no other messages are getting handled during negotiation. Perhaps we can change this in the future.